vllm/engine/output_processor/interfaces.py

```diff
@@ -30,13 +30,14 @@ def create_output_processor(
         seq_counter: Counter,
         get_tokenizer_for_seq: Callable[[Sequence], AnyTokenizer],
         stop_checker: "StopChecker",
+        force_single_step: bool = False,
     ):
         """Create an output processor.
 
         This returns a single-step output processor if num_lookahead_slots is
         zero, else returns a multi-step output processor.
         """
-        if scheduler_config.num_lookahead_slots == 0:
+        if scheduler_config.num_lookahead_slots == 0 or force_single_step:
```

Comment:
Does that mean we are wasting lookahead slots when forcing single step?

Reply:
fair; fixed. `num_lookahead_slots` in `SchedulerConfig` is now a property which is forced to zero when multi-step is disabled in the current step:

```
    @property
    def num_lookahead_slots(self) -> int:
        """Getter method that gets num_lookahead_slots

        Force to zero if multi-step is disabled in the current step.
        """
        return (self._num_lookahead_slots
                if self.current_step_is_multi_step else 0)

    @num_lookahead_slots.setter
    def num_lookahead_slots(self, value: int) -> None:
        """Setter method for num_lookahead_slots"""
        self._num_lookahead_slots = value
```

Reply:
LGTM. Then could we avoid to introduce `force_single_step` in this interface?